src/acuvity/apexdiscovery.py

```diff
@@ -56,13 +60,17 @@ def well_known_apex_info(client: HttpClient, token: str, url: str, iteration: in
             raise ValueError("Too many redirects")
         req = client.build_request("GET", url, headers={"Authorization": f"Bearer {token}"})
         resp = client.send(req, follow_redirects=False) # following redirects automatically will remove the token from the call as headers are not going to be sent anymore
+
+        if resp.status_code == 401:
+            raise ValueError("Unauthorized: Invalid token or insufficient permissions")
+
         if resp.is_redirect:
             return well_known_apex_info(client, token, resp.headers["Location"], iteration + 1)
         return resp.json()
     try:
         apex_info = well_known_apex_info(client, token, f"{api_url}/.well-known/acuvity/my-apex.json")
     except Exception as e:
-        raise ValueError("Failed to get apex info from well-known endpoint") from e
+        raise ValueError(f"Failed to get apex info from well-known endpoint: {str(e)}") from e
```

Comment:
that's why I've put the `from e` because the error in there feels redundant. But if you feel strongly about this, we can keep that.